java/gateway-core/src/main/java/com/avaulta/gateway/pseudonyms/impl/UrlSafeTokenPseudonymEncoder.java

```diff
@@ -43,9 +44,13 @@ public class UrlSafeTokenPseudonymEncoder implements PseudonymEncoder {
     public String encode(Pseudonym pseudonym) {
         String encoded;
         if (pseudonym.getReversible() == null) {
-            encoded = TOKEN_PREFIX + encoder.encodeToString(pseudonym.getHash());
+            encoded = TOKEN_PREFIX + encoder.encodeToString(pseudonym.getHash());;
         } else {
-            encoded = REVERSIBLE_PREFIX + encoder.encodeToString(pseudonym.getReversible());
+            if (!Arrays.equals(pseudonym.getHash(), 0, pseudonym.getHash().length, pseudonym.getReversible(), 0, pseudonym.getHash().length)) {
```

Comment:
:red_circle: is this correct?? last byte of the hash can be padding, so it may not match exactly.
from your tests
`        "123.234.252.12,t~7USliSM4GiS0Xfk1DXIAH-4nK-UkLJlSAA_5ZqQh_CI", `
`        "123.234.252.12,p~7USliSM4GiS0Xfk1DXIAH-4nK-UkLJlSAA_5ZqQh_CJStJlrfaIPHLZUvx-dHHQp",`

7USliSM4GiS0Xfk1DXIAH-4nK-UkLJlSAA_5ZqQh_**CI** != 7USliSM4GiS0Xfk1DXIAH-4nK-UkLJlSAA_5ZqQh_**CJ**

Reply:
that's just an artifact of the encoding. Each char of base64 encodes 6 bits of the binary sequence; 32 byte hash --> 43 bytes of base64

in 2nd case, you take 32 bytes + X bytes of encrypted content, and THEN base64 whole thing. so some bits from encrypted content alter the 43rd byte of the base64 of that.